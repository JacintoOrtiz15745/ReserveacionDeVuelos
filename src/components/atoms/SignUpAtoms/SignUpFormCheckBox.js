import React from 'react';
import {View, Text} from 'react-native';
import {styles} from '../../../styles/SignUpStyles';
import CheckBox from '@react-native-community/checkbox';
const SignUpFormCheckBox = (props) => {
  const {
    checkboxMessage,
    checkBoxValue,
    formValues,
    setFormValues,
    handleChange,
  } = props;
  return (
    <View style={styles.checkBoxContainer} testID='view'>
      <CheckBox
        testID='checkBox'
        value={checkBoxValue}
        onValueChange={(value) =>
          handleChange(formValues, setFormValues, value)
        }
      />
      <Text>{checkboxMessage}</Text>
    </View>
  );
};

export default SignUpFormCheckBox;
