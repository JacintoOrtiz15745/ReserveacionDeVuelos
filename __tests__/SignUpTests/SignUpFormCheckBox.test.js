import React from 'react';
import SignUpFormCheckBox from '../../src/components/atoms/SignUpAtoms/SignUpFormCheckBox';
import {shallow} from 'enzyme';

const formValues = {
  name: '',
  email: '',
  password: '',
  termsCheckbox: false,
  updatesCheckbox: false,
  validName: true,
  validEmail: true,
  validPassword: true,
  validatedName: false,
  validatedEmail: false,
  validatedPassword: false,
};
const props = {
  checkboxMessage: SignUpFormCheckBox.checkboxMessage,
  checkBoxValue: false,
  formValues,
  setFormValues: jest.fn(),
  handleChange: jest.fn(),
};

describe('Reders conrreclty', () => {
  const wrapper = shallow(<SignUpFormCheckBox {...props} />);
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Render correctly', () => {
    const render = wrapper.findWhere((node) => node.prop('testID') === 'view');
    expect(render.exists()).toBeTruthy();
  });
  test('Render only one checkbox', () => {
    const render = wrapper.findWhere(
      (node) => node.prop('testID') === 'checkBox',
    );
    expect(render.length).toBe(1);
  });
  test('Render by default checkbox on false', () => {
    const value = wrapper.findWhere(
      (node) => node.prop('testID') === 'checkBox',
    );

    expect(value.props().value).toBe(false);
  });
  test('Render correctly the event onValueChange', () => {
    const event = wrapper.findWhere(
      (node) => node.prop('testID') === 'checkBox',
    );
    event.props().onValueChange();
    wrapper.update();
    expect(props.handleChange).toHaveBeenCalled();
  });
});
