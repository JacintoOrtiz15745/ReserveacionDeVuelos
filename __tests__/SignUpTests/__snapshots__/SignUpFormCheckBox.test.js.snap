// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reders conrreclty snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginTop": 5,
    }
  }
  testID="view"
>
  <ForwardRef(CheckBoxWithRef)
    onValueChange={[Function]}
    testID="checkBox"
    value={false}
  />
  <Text />
</View>
`;
